src/app/modules/shared/models/constraints/combination-constraint.ts

```diff
@@ -0,0 +1,33 @@
+import {Constraint} from './constraint';
+
+export class CombinationConstraint implements Constraint {
+
+  private children: Constraint[];
+  public NOT: boolean;
+  public AND: boolean;
+  public OR: boolean;
```

Comment:
As these attributes are not constants, I'd rather not use all-caps for the name. Also, aren't AND and OR mutually exclusive? And as they are booleans, maybe isNot and isAnd would be better names, and the OR would be deduced from isAnd.